components/map/basic/src/index.js

```diff
@@ -0,0 +1,227 @@
+   */
+  styles: PropTypes.array,
+  /**
+   * An object with the definition of our map. All options of here maps are allowed, for example
+   * { zoom: <number>, center: { lat: number, lng: number} }
+   */
+  options: PropTypes.object.isRequired,
+  /**
+   * Our appid token, provided by HERE maps
+   */
+  appId: PropTypes.string.isRequired,
+  /**
+   * Our appCode token, provided by HERE maps
+   */
+  appCode: PropTypes.string.isRequired,
+  /**
+   * If is a interactable map (drag, zoom)... With this enabled the UI will be automatically loaded
+   */
+  isInteractive: PropTypes.bool,
+  /**
+   * Use CIT ONLY SHOULD BE ENABLED if we are going to use a customer integration testing app id and app code
+   * More info about CIT here https://developer.here.com/documentation/venue-maps/topics/request-cit-environment.html
+   */
+  useCIT: PropTypes.bool,
+  /**
+   * Enables or disables HTTPS communication with the platform.
+   */
+  useHTTPS: PropTypes.bool,
+  /**
+   * An object with the definiton of our markers. The icons could be a PLAIN STRING SVG definition or a path to an svg or image.
+   * Accepts an array of positions to position multiple markers on the map asswell as a custom icon on each marker position.
+   * If no icon is provided on each element the component will fallback to the genericIcon.
+   */
+  marker: PropTypes.shape({
+    /**
+     * A STRING DEFINITION of an SVG or a relative path to a file
+     */
+    genericIcon: PropTypes.string.isRequired,
+    elements: PropTypes.arrayOf(React.PropTypes.shape({
+      lat: PropTypes.number.isRequired,
+      lng: PropTypes.number.isRequired,
+      icon: PropTypes.string
+    }))
+  }),
+  /**
+   * An object with the definition of our 'radar' circle.
+   * The style could be generic (applied to all the radars on the map) or custom by each element.
+   * The element accept a lattitude a longitude a radius and a custom styling prop.
+   * If no styling prop is provided on the element component will fallback to styleGeneric.
+   */
+  circleShape: PropTypes.shape({
+    styleGeneric: PropTypes.shape({
+      strokeColor: PropTypes.string,
+      fillColor: PropTypes.string,
+      lineWidth: PropTypes.number,
+      lineCap: PropTypes.string,
+      lineJoin: PropTypes.string
+    }),
+    elements: PropTypes.arrayOf(React.PropTypes.shape({
+      lat: PropTypes.number.isRequired,
+      lng: PropTypes.number.isRequired,
+      radius: PropTypes.number.isRequired,
+      style: PropTypes.shape({
+        strokeColor: PropTypes.string,
+        fillColor: PropTypes.string,
+        lineWidth: PropTypes.number,
+        lineCap: PropTypes.string,
+        lineJoin: PropTypes.string
+      })
+    }))
+  })
+}
+
+MapBasic.defaultProps = {
+  libraries: ['http://js.api.here.com/v3/3.0/mapsjs-core.js', 'http://js.api.here.com/v3/3.0/mapsjs-service.js', 'https://js.cit.api.here.com/v3/3.0/mapsjs-ui.js', 'https://js.cit.api.here.com/v3/3.0/mapsjs-mapevents.js'],
```

Comment:
I have several this to say about it:
1. Some of the libs are with the http protocol, which would provoke problems in https pages
A good practice is to define urls os CDNs protocol relative like this:
```js
'//js.api.here.com/v3/3.0/mapsjs-core.js', 
'//js.api.here.com/v3/3.0/mapsjs-service.js'
'//js.cit.api.here.com/v3/3.0/mapsjs-ui.js'
'//js.cit.api.here.com/v3/3.0/mapsjs-mapevents.js'
```

2. Your are checking the libs loaded on every single instance obtaining them from the DOM.
You easily take track of if it's the first instance or not and do theses actions ONLY THE FIRST TIME. It's a matter or performance.

3. You are add scripts in the page, multiple requests. Have you considered using the npm version? https://www.npmjs.com/package/here-js-api
You could delay the loading of the library with require.ensure I guess.